src/main/java/org/interledger/cryptoconditions/Fulfillment.java

```diff
@@ -1,13 +1,36 @@
 package org.interledger.cryptoconditions;
 
-public interface Fulfillment {
+/**
+ * An implementation of a crypto-conditions Fulfillment.
+ *
+ * @see "https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/"
+ */
+public interface Fulfillment<C extends Condition> {
 
-  ConditionType getType();
+  /**
+   * Accessor for the type of this fulfillment.
+   *
+   * @return A {@link CryptoConditionType} for this fulfillment.
+   */
+  CryptoConditionType getType();
 
-  byte[] getEncoded();
```

Comment:
Nah, you're suggestion is probably the correct way to do this. A new class called `CryptoConditionKey` or something, when we need it, one day